src/equation_systems/equation_system.C

```diff
@@ -269,6 +316,7 @@ EquationSystem::BuildBilinearForms()
 
       for (auto & blf_kernel : blf_kernels)
       {
+        blf->SetAssemblyLevel(_assembly_level);
```

Comment:
This will set the assembly level multiple times for the same bilinear form - likely should be outside this loop?

Reply:
Indeed it should be outside of the loop. Fixed!